src/lib/openjp3d/tcd.c

```diff
@@ -226,10 +226,10 @@ void tcd_malloc_encode(opj_tcd_t *tcd, opj_volume_t * volume, opj_cp_t * cp,
         if (tcp->rates[j] <= 1) {
             tcp->rates[j] = 0;
         } else {
-            float num = (float)(tile->numcomps * (tile->x1 - tile->x0) *
-                                (tile->y1 - tile->y0) * (tile->z1 - tile->z0) * volume->comps[0].prec);
-            float den = (float)(8 * volume->comps[0].dx * volume->comps[0].dy *
-                                volume->comps[0].dz);
+            float num = (float)tile->numcomps * (tile->x1 - tile->x0) *
+                                (tile->y1 - tile->y0) * (tile->z1 - tile->z0) * volume->comps[0].prec;
```

Comment:
That avoids an overflow, but causes several float operations, so will be slower. Is this acceptable here?

Which part of the integer operations did overflow? Would it help to use larger integer types?